base_layer/contacts/src/contacts_service/storage/types/contacts.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::convert::TryFrom;
+
+use chrono::NaiveDateTime;
+use diesel::{prelude::*, SqliteConnection};
+use tari_common_sqlite::util::diesel_ext::ExpectedRowsExtension;
+use tari_common_types::tari_address::TariAddress;
+use tari_comms::peer_manager::NodeId;
+use tari_utilities::ByteArray;
+
+use crate::{
+    contacts_service::{error::ContactsServiceStorageError, types::Contact},
+    schema::contacts,
+};
+
+/// A Sql version of the Contact struct
+#[derive(Clone, Debug, Queryable, Insertable, PartialEq, Eq)]
+#[diesel(table_name = contacts)]
+pub struct ContactSql {
+    pub address: Vec<u8>,
+    node_id: Vec<u8>,
```

Comment:
Whats the reason we have both here?
address = [node_id][CRC + network]
Asking the node id from address is a simple slice read